spring-ai-core/src/main/java/org/springframework/ai/model/tool/ToolCallingManager.java

```diff
@@ -39,7 +38,7 @@ public interface ToolCallingManager {
 	/**
 	 * Execute the tool calls requested by the model.
 	 */
-	List<Message> executeToolCalls(Prompt prompt, ChatResponse chatResponse);
+	ToolExecutionResult executeToolCalls(Prompt prompt, ChatResponse chatResponse);
```

Comment:
I introduced this to propagate back whether the "returnDirect" option is enabled. Even without that use case, I think it's still a better return type than what I used before (`List<Message>`). It's more easily understandable.